test/tests/functional/basic_single_server_auth_tests.js

```diff
@@ -7,29 +7,37 @@ var expect = require('chai').expect,
   Connection = require('../../../lib/connection/connection'),
   Bson = require('bson');
 
-// Skipped due to use of topology manager
+const MongoCredentials = require('../../../lib/auth/mongo_credentials').MongoCredentials;
+
 describe('Basic single server auth tests', function() {
   it('should correctly authenticate server using scram-sha-256 using connect auth', {
     metadata: { requires: { topology: 'auth', mongodb: '>=3.7.3' } },
     test: function(done) {
       const config = this.configuration;
-      const method = 'scram-sha-256';
-      const user = 'user';
+      const mechanism = 'scram-sha-256';
+      const source = 'admin';
+      const username = 'user';
       const password = 'pencil';
       const createUserCommand = {
-        createUser: user,
+        createUser: username,
         pwd: password,
-        roles: [{ role: 'root', db: 'admin' }],
+        roles: [{ role: 'root', db: source }],
         digestPassword: true
       };
-      const dropUserCommand = { dropUser: user };
-      const auth = [method, 'admin', user, password];
+      const dropUserCommand = { dropUser: username };
+
+      const credentials = new MongoCredentials({
```

Comment:
question: these can all be expressed by connection string, shouldn't we use that? Follow up question: should we provide a means of extracting `MongoCredentials` from a connection string, and/or simply have the connection string parser output credentials for the `auth` object?

Reply:
Final thought: this might be the case for not using the key `credentials`. The URI parser, by spec, parses into an `auth` key so we'd have to do translation there. Food for thought.

Reply:
I've added helper functions in [the native PR](https://github.com/mongodb/node-mongodb-native/pull/1826) to do this. Rather than parse auth credentials out of the connection string, I have a function that parses the output of the url parser into credentials. My thinking is that, since native supports any combination of url and natively passed options, it is better to wait until both url options and natively passed options are merged.

Reply:
SGTM